core/src/main/scala/org/bitcoins/core/crypto/HashDigest.scala

```diff
@@ -109,49 +106,44 @@ object Sha256DigestBE extends Factory[Sha256DigestBE] {
 /**
   * Represents the result of SHA256(SHA256())
   */
-sealed trait DoubleSha256Digest extends Any with HashDigest {
-  def flip: DoubleSha256DigestBE = DoubleSha256DigestBE(bytes.reverse)
+case class DoubleSha256Digest(bytes: ByteVector) extends HashDigest {
```

Comment:
This causes runtime allocations that were optimized in #690

Reply:
You can't have `require()` in the case class body if you want to avoid allocating https://docs.scala-lang.org/overviews/core/value-classes.html

Reply:
We could make `DoubleSha256Digest` a `AnyVal`, but that gives us a compile error in `BloomFilter`, saying that `DoubleSha256Digtest` and `ByteVector` are the same type after type erasure. What's your thoughts on removing `BloomFilter.insert(DoubleSha256Digest)`?

Reply:
So in general, this stuff needs to be addressed in #717. I think it is probably out of scope of this PR unless it really helps the test coverage story too?

Reply:
Might help a bit with coverage, but not much. Agree on out of scope.